src/components/join-flow/join-flow.jsx

```diff
@@ -23,6 +23,7 @@ class JoinFlow extends React.Component {
         super(props);
         bindAll(this, [
             'handleAdvanceStep',
+            'handleMidRegistrationError',
```

Comment:
Could we just call this "handle/onRegistrationError" rather than "MidRegistration"? I know this isn't only used for errors during the final registration request, but I think this could be more broadly applicable.

Reply:
Yes. Done! :)